internal/services/containers/container_registry_task_resource.go

```diff
@@ -0,0 +1,1573 @@
+	FileTaskStep        []FileTaskStep         `tfschema:"file_task_step"`
+	EncodedTaskStep     []EncodedTaskStep      `tfschema:"encoded_task_step"`
+	BaseImageTrigger    []BaseImageTrigger     `tfschema:"base_image_trigger"`
+	SourceTrigger       []SourceTrigger        `tfschema:"source_trigger"`
+	TimerTrigger        []TimerTrigger         `tfschema:"timer_trigger"`
+	RegistryCredential  []RegistryCredential   `tfschema:"registry_credential"`
+	Tags                map[string]interface{} `tfschema:"tags"`
+}
+
+func userDataStateFunc(v interface{}) string {
+	switch s := v.(type) {
+	case string:
+		return utils.Base64EncodeIfNot(s)
+	default:
+		return ""
+	}
+}
+
+func (r ContainerRegistryTaskResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryTaskName,
+		},
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RegistryID,
+		},
+		"platform_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"os": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Windows),
+							string(legacyacr.Linux),
+						}, false),
+					},
+					"architecture": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.Amd64),
+							string(legacyacr.Arm),
+							string(legacyacr.Arm64),
+							string(legacyacr.ThreeEightSix),
+							string(legacyacr.X86),
+						}, false),
+					},
+					"variant": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(legacyacr.V6),
+							string(legacyacr.V7),
+							string(legacyacr.V8),
+						}, false),
+					},
+				},
+			},
+		},
+		"docker_step": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"dockerfile_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"context_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"context_access_token": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"image_names": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"is_push_enabled": {
```

Comment:
we don't use is in boolean properties?
```suggestion
					"push_enabled": {
```